app/validators/placeholders/placeholder_validator.py

```diff
@@ -108,6 +108,19 @@ def validate_option_label_from_value_placeholder(self, answer_id):
                 identifier=answer_id,
             )
 
+    def validate_answer_type_correct(self, argument, transform_type):
+        if argument.get("source") == "answers":
+            transform_type = transform_type.split("_")[1]
+            answer_id = argument.get("identifier")
+            answer_type = self.questionnaire_schema.answers_with_context[answer_id][
+                "answer"
+            ]["type"]
+            if answer_type.lower() != transform_type:
```

Comment:
For units, we need to also validate the `unit` itself matches. The answer cant set unit to `length-meter` and then have transformed state `length-mile`, they should match.

Reply:
Ok, added extra method of validation to check for unit mismatches.